build-scripts/rcm-guest/push-to-mirrors.sh

```diff
@@ -68,28 +69,45 @@ LASTDIR=$(readlink ${PUDDLEDIR})
 
 echo "Pushing puddle: $LASTDIR"
 
-# Copy all files from the last latest into a directory for the new puddle (jmp: in order to prevent as much transfer as possible by rysnc for things which weren't rebuilt?)
-ssh ${BOT_USER} -o StrictHostKeychecking=no use-mirror-upload.ops.rhcloud.com "cd /srv/enterprise/${REPO} ; cp -r --link latest/ $LASTDIR"
-
-# Copy the local puddle to the new, remote location.
-rsync -aHv --delete-after --progress --no-g --omit-dir-times --chmod=Dug=rwX -e "ssh ${BOT_USER} -o StrictHostKeyChecking=no" ${PUDDLEDIR}/ use-mirror-upload.ops.rhcloud.com:/srv/enterprise/${REPO}/${LASTDIR}/
+MIRROR_SSH_SERVER="use-mirror-upload.ops.rhcloud.com"
+MIRROR_SSH_BASE="ssh ${BOT_USER} -o StrictHostKeychecking=no"
+MIRROR_SSH="${MIRROR_SSH_BASE} ${MIRROR_SSH_SERVER}"
+MIRROR_PATH="/srv/enterprise/${REPO}"
+ALL_DIR="/srv/enterprise/all/${MAJOR_MINOR}"
 
-# Replace latest link with new puddle content
-ssh ${BOT_USER} -o StrictHostKeychecking=no use-mirror-upload.ops.rhcloud.com "cd /srv/enterprise/${REPO} ; ln -sfn $LASTDIR latest"
+$MIRROR_SSH sh -s <<-EOF
+  set -e
```

Comment:
IMO here-docs with code in them are an anti-pattern. You don't get syntax highlighting in there, you have to worry about where and how variables expand, etc. If it's possible, maybe it would be cleaner to get all the remote logic in one place (forward params as necessary?) and use a separate file you can `scp` over and execute on the remote host.